pulp_smash/tests/rpm/api_v2/test_force_full.py

```diff
@@ -0,0 +1,200 @@
+            gen_distributor(),
+        ).json()
+        return (repo, distributor)
+
+    @classmethod
+    def setUpClass(cls):
+        """Create repository, associate, publish, associate, publish, publish.
+
+        Following steps are executed:
+
+        1. Create repository 1 with feed, sync and publish it.
+        2. Create repository 2 with feed
+        3. Copy 3 random units from repository 1 to repository 2
+        4. Publish repository 2
+        5. Copy 3 random units from repository 1 to repository 2
+        6. Publish repository 2
+        7. Publish repository 2 with force_full
+        """
+        super(ForceFullTestCase, cls).setUpClass()
+        cls.responses = {}
+        client = api.Client(cls.cfg)
+
+        # Create repo 1
+        repo1, _ = cls.setup_repo(client)
+        cls.resources.add(repo1['_href'])  # mark for deletion
+
+        # Sync repo 1
+        cls.responses['sync'] = utils.sync_repo(cls.cfg, repo1['_href'])
+        report = cls.responses['sync'].json()
+        next(api.poll_spawned_tasks(cls.cfg, report))
+
+        # Get all units
+        cls.responses['repo1_units'] = client.post(
+            urljoin(repo1['_href'], 'search/units/'),
+            {'criteria': {'type_ids': ['rpm']}},
+        ).json()
+
+        # Create repo 2
+        repo2, distributor2 = cls.setup_repo(client)
+        cls.resources.add(repo2['_href'])  # mark for deletion
+
+        rpm_units = cls.responses['repo1_units']
+
+        # copy 3 random units
+        units_to_copy = []
+        for _ in range(0, 3):
+            units_to_copy.append(rpm_units.pop(0))
+
+        report = _copy_units(cls.cfg, repo2['_href'],
+                             repo1['id'], units_to_copy)
+        next(api.poll_spawned_tasks(cls.cfg, report))
+
+        cls.responses['repo2_publish1'] = client.post(
+            urljoin(repo2['_href'], 'actions/publish/'),
+            {'id': distributor2['id']},
+        ).json()
+        next(api.poll_spawned_tasks(cls.cfg, cls.responses['repo2_publish1']))
+
+        # copy another 3 random units
+        units_to_copy2 = []
+        for _ in range(0, 3):
+            units_to_copy2.append(rpm_units.pop(0))
+
+        report = _copy_units(cls.cfg, repo2['_href'],
+                             repo1['id'], units_to_copy2)
+        wait_for_tasks(client, cls.cfg, [report])
+
+        cls.responses['repo2_publish2'] = client.post(
+            urljoin(repo2['_href'], 'actions/publish/'),
+            {'id': distributor2['id']},
+        ).json()
+        wait_for_tasks(client, cls.cfg, [cls.responses['repo2_publish2']])
+
+        # publish with force_full
+        cls.responses['repo2_publish_full'] = client.post(
+            urljoin(repo2['_href'], 'actions/publish/'),
+            {'id': distributor2['id'], 'force_full': True},
+        ).json()
+
+    def test_initial_publish(self):
+        """Verify the HTTP status code of each publish."""
```

Comment:
The docstrings on these methods don't match what the methods do. I'll re-write these docstrings to reflect the actual assertions, and if possible, I'll add an explanation of why these assertions are used. This process is error-prone, though, and the end result is more likely to be correct if the docstrings are correct.